primaryHandler.go

```diff
@@ -563,26 +563,31 @@ func validateDeviceID() alice.Chain {
 	})
 }
 
-func ValidateWRP() func(http.Handler) http.Handler {
+func ValidateWRP(logger *zap.Logger) func(http.Handler) http.Handler {
 	return func(delegate http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 
 			if msg, ok := wrpcontext.GetMessage(r.Context()); ok {
-				validators := wrp.SpecValidators()
 				var err error
+
+				validators := wrp.SpecValidators()
 				for _, v := range validators {
 					err = multierr.Append(err, v.Validate(*msg))
 				}
+
 				if err != nil {
```

Comment:
I don't think we need this `nil` check since  errors.Is should handle this
```suggestion
				if err != nil {
```